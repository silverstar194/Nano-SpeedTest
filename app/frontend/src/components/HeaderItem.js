import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const HeaderItem = ({activeTab, onSwitchTab, tabName, to, text}) => {
	const styles = 'nav-link ' + (activeTab === tabName ? 'active' : '');
    return (
      	<li className="nav-item">
			<Link className={styles} to={to} onClick={() => onSwitchTab(to)}> {text} </Link>
      	</li>
    )
};

HeaderItem.propTypes = {
	activeTab: PropTypes.string.isRequired,
	onSwitchTab: PropTypes.func.isRequired,
	tabName: PropTypes.string.isRequired,
	to: PropTypes.string.isRequired,
	text: PropTypes.string.isRequired
}

export default HeaderItem;