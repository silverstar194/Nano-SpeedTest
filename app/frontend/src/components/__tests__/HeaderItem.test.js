import React from 'react';
import renderer from 'react-test-renderer';
import HeaderItem from '../HeaderItem';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter} from 'react-router-dom';
import { shallow } from 'enzyme';

const mockStore = configureStore();
let store = mockStore({
});

const homeProps = {
    activeTab: '',
	onSwitchTab: () => undefined,
	tabName: '',
	to: '',
	text: 'Speed Test'
};

it('shallow renders without crashing', () => {
    shallow(<HeaderItem {...homeProps}/>);
});

it('renders homepage correctly', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <HeaderItem {...homeProps}  />
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

const statsProps = {
    activeTab: '',
	onSwitchTab: () => undefined,
    to:'Stats',
    text:'Statistics',
    tabName:'Stats'
};

it('renders stats page correctly even if not active tab', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <HeaderItem {...statsProps}  />
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });