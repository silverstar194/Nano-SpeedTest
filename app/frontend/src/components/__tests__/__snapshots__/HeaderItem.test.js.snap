// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage correctly 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link active"
    href="/"
    onClick={[Function]}
  >
     
    Speed Test
     
  </a>
</li>
`;

exports[`renders stats page correctly even if not active tab 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link "
    href="/Stats"
    onClick={[Function]}
  >
     
    Statistics
     
  </a>
</li>
`;
